doc/nrf/doc_build_process.rst

```diff
@@ -0,0 +1,78 @@
+.. _doc_build_process:
+
+Documentation build process
+###########################
+Whenever the main branch of the `sdk-nrf`_ repository is updated or a pull request is made, a new version of the documentation is built and published. The following section describes the steps involved with building and publishing the documentation.
+
+Building with Sphinx
+********************
+The |NCS| documentation uses the Sphinx framework to build the docsets for HTML.
+Sphinx primarily uses three files to work:
+
+* A makefile that runs the :code:`sphinx-build` command in the correct folder. This folder should contain the next two files.
+* A file :file:`index.rst` that serves as the top of the ToC tree and the root of the documentation.
+* A file :file:`conf.py` in which all further configuration of Sphinx is placed.
+
+The |NCS| documentation does not use a makefile for building Sphinx, and instead uses Ninja. Ninja is a faster version of makefile with less features and complexity, but :file:`.ninja` files are not designed to be hand-written.
+Instead, CMake is used to create a :file:`build.ninja` file, which in turn can build the documentation.
+
+CMake
+-----
+CMake not only creates the ninja file used to build the documentation, it also checks for the required binaries and dependencies and ensures that the docsets are built in the correct order.
+:file:`nrf/doc/CMakeLists.txt` contains the function :code:`add_docset` which is used to create targets for all the docsets.
+This includes the :code:`{docset}`/ :code:`{docset}-all`, :code:`{docset}-inventory`/ :code:`{docset}-inventory-all`, :code:`{docset}-linkcheck` and :code:`{docset}-clean` targets.
+The inventory targets create reference targets without building the documentation, providing a solution for any circular dependencies.
+Some docsets require additional source files to be generated before the docset can be built, like :ref:`Zephyr's device tree<devicetree-intro>`.
+In these cases a separate target is created and added as a dependency for the docset target.
+
+External contents
+-----------------
+When the :code:`ninja` command executes :code:`sphinx-build`, a :file:`_build/{docset}/src` directory is used as the source directory.
```

Comment:
I'd avoid mentions to `ninja`, I think it's not relevant here. also, you can use for literals:

```rst
``literal``
```